shoop/simple_supplier/templates/shoop/simple_supplier/admin/product_form_part.jinja

```diff
@@ -1,26 +1,59 @@
 {% from "shoop/admin/macros/general.jinja" import content_block with context %}
 {% set ss_form = form["simple_supplier"] %}
-{% set product_qty = ss_form.products|length %}
-{% if product_qty %}
-    {% call content_block(_("Stock management"), "fa-cubes") %}
-        {% set product_qty = ss_form.products|length %}
-        {% for product in ss_form.products %}
-            {% set suppliers = ss_form.get_suppliers(product) %}
-            {% if suppliers and product_qty > 1 %}<h2>{{ product.name }}</h2>{% endif %}
-            {% for supplier in suppliers %}
-                <h3>{{ supplier.name }}</h3>
-                <div class="row">
-                    <div class="col-md-6">
-                        {{ ss_form.get_stock_information(supplier, product)|safe }}
-                    </div>
-                    <div class="col-md-6">
-                        {{ ss_form.get_stock_adjustment_form(supplier, product)|safe }}
-                    </div>
+
+{%  macro render_product(suppliers, product, product_url, product_qty) %}
+    {% if suppliers %}
+        {% if product_qty > 1 %}<h2><a href="{{ product_url }}" target="_blank">{{ product.name }}</a></h2>{% endif %}
+        {% for supplier in suppliers %}
+            <h3>{{ supplier.name }}</h3>
+                <div class="col-md-6">
+                    {{ ss_form.get_stock_information(supplier, product)|safe }}
+                </div>
+                <div class="col-md-6">
+                    {{ ss_form.get_stock_adjustment_form(supplier, product)|safe }}
                 </div>
-            {% endfor %}
         {% endfor %}
-    {% endcall %}
-{% endif %}
+    {% else %}
+        <p>
+            {% trans product_name=product.name, product_url=product_url, module_name=ss_form.module_name %}
```

Comment:
Translatable strings shouldn't have leading and trailing space. You can have Jinja get rid of those for you by adding a dash in the surrounding tag, i.e. `{% trans ... %}` would be `{% trans ... -%}`.